src/components/delegate/authorize-claimer-form.tsx

```diff
@@ -0,0 +1,74 @@
+import { RVR_AUTHORIZER } from '@/constants/contracts'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { QueryKey, useQueryClient } from '@tanstack/react-query'
+import { Check } from 'lucide-react'
+import { useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import { isAddress } from 'viem'
+import { useWaitForTransactionReceipt, useWriteContract } from 'wagmi'
+import { z } from 'zod'
+import { Button } from '../ui/button'
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from '../ui/form'
+import { Input } from '../ui/input'
+
+const formSchema = z.object({
+  address: z.custom((value) => typeof value === 'string' && isAddress(value), {
+    message: 'Invalid address',
+  }),
+})
+
+type AuthorizeClaimerFormProps = {
+  authorizedClaimerQueryKey: QueryKey
+}
+
+export const AuthorizeClaimerForm = ({ authorizedClaimerQueryKey }: AuthorizeClaimerFormProps) => {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  })
+
+  const qc = useQueryClient()
+  const { data: hash, writeContract, isPending } = useWriteContract()
+
+  const { isLoading: isConfirming, isSuccess: isConfirmed } = useWaitForTransactionReceipt({
+    hash,
+  })
+
+  useEffect(() => {
+    if (isConfirmed) {
+      qc.invalidateQueries({ queryKey: authorizedClaimerQueryKey })
+    }
+  }, [authorizedClaimerQueryKey, isConfirmed, qc])
+
+  function onSubmit(form: z.infer<typeof formSchema>) {
+    writeContract({
+      address: RVR_AUTHORIZER.sepolia,
```

Comment:
The contract used should follow from the network that the user is connected to rather than be hard coded. Right now, I can switch to Mainnet in the ui, but my authorized claimer will invariably be transacted against the wrong contract.